crates/fuel-core/src/service/sub_services.rs

```diff
@@ -160,14 +178,21 @@ pub fn init_sub_services(
     #[cfg(not(feature = "p2p"))]
     let p2p_adapter = P2PAdapter::new();
 
-    let gas_price_provider = StaticGasPrice::new(config.static_gas_price);
+    let old_gas_price_provider = StaticGasPrice::new(config.static_gas_price);
+
+    let update_algo = StaticAlgorithmUpdater::new(config.static_gas_price);
+    let gas_price_task = GasPriceService::new(last_height, update_algo);
+    let next_algo = gas_price_task.next_block_algorithm();
+    let gas_price_service = ServiceRunner::new(gas_price_task);
```

Comment:
It would be nice if you followed pattern from other services. Usually each service defined `new_service` function that is doing everything inside. If you need access to shared state, you can call `let also = service.shared_state()`.